ipatests/test_integration/test_backup_and_restore.py

```diff
@@ -312,13 +316,22 @@ def _full_backup_restore_with_DNS_zone(self, reinstall=False):
                                      '--uninstall',
                                      '-U'])
 
+            self.master.run_command([pkg_installer, '-y',
+                                     'remove', '*ipa-server-dns'])
             if reinstall:
                 tasks.install_master(self.master, setup_dns=True)
 
             dirman_password = self.master.config.dirman_password
+            result = self.master.run_command(
+                ['ipa-restore', backup_path],
+                stdin_text=dirman_password + '\nyes',
+                raiseonerr=False)
+            assert 'Please install the package' in result.stderr_text
+
+            self.master.run_command([pkg_installer, '-y',
+                                     'install', '*ipa-server-dns'])
```

Comment:
The method tasks.install_packages(self.master, ['*ipa-server-dns']) can be used instead.